src/proof_production/Contradiction.h

```diff
@@ -0,0 +1,35 @@
+/*********************                                                        */
+/*! \file Contradiction.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Omri Isac, Guy Katz
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2022 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+ **/
+
+#ifndef __Contradiction_h__
+#define __Contradiction_h__
+
+/*
+  Contains all info relevant for a simple Marabou contradiction - i.e. explanations of contradicting bounds of a variable
+*/
+class Contradiction
+{
+public:
+    Contradiction( unsigned var, double *upperBoundExplanation, double *lowerBoundExplanation );
+    ~Contradiction();
+
+private:
+    unsigned _var;
+    double *_upperBoundExplanation;
+    double *_lowerBoundExplanation;
+
+friend class UnsatCertificateNode;
```

Comment:
We try to avoid using friend classes; I think there's not a single one in the code base. Can this be circumvented?